messaging/WFLY-20512_expose_scale_down_commit_interval.adoc

```diff
@@ -0,0 +1,136 @@
+  - messaging
+stability-level: default
+issue: https://github.com/wildfly/wildfly-proposals/issues/709
+
+feature-team:
+ developer: ehsavoie
+ sme:
+  -
+ outside-perspective:
+  -
+---
+
+= Expose Artemis commit-interval attribute for scaledown
+:author:            Emmanuel Hugonnet
+:email:             ehugonne@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+When moving messages from the server scaling down, all the operation is done in a single transaction which means that all the messages need to be in memory and can lead to a OOME. The purpose of this feature is to enable the size of the transaction (aka the number of messages) to be defined so that messages are moved by blocks instead of a single block.
+
+=== User Stories
+
+As an administrator I want to be able to scale down a node in my cluster when the load is decreasing and configuring the number of messages to be moved  in a batch.
+
+== Issue Metadata
+
+=== Related Issues
+ * https://issues.redhat.com/browse/EAP7-2296[EAP7-2296]
+ * https://issues.redhat.com/browse/JBEAP-28739[JBEAP-28739]
+ * https://issues.redhat.com/browse/WFLY-19752[WFLY-19752]
+ * https://issues.apache.org/jira/browse/ARTEMIS-5053[ARTEMIS-5053]
+ 
+
+=== Affected Projects or Components
+
+* https://github.com/apache/activemq-artemis/[Apache Artemis]
+* WildFly
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* Managed domain
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+
+* Add an optional `scale-down-commit-interval` integer to the  https://docs.wildfly.org/35/wildscribe/subsystem/messaging-activemq/server/ha-policy=replication-colocated/configuration=secondary/index.html[/subsystem=messaging-activemq/server/ha-policy=replication-colocated/configuration=secondary] to control the size of the messages transaction block.
+* Add an optional `scale-down-commit-interval` integer to the  https://docs.wildfly.org/35/wildscribe/subsystem/messaging-activemq/server/ha-policy=shared-store-colocated/configuration=secondary/index.html[/subsystem=messaging-activemq/server/ha-policy=shared-store-colocated/configuration=secondary] to control the size of the messages transaction block.
+* Add an optional `scale-down-commit-interval` integer to the  https://docs.wildfly.org/35/wildscribe/subsystem/messaging-activemq/server/ha-policy=shared-store-secondary/index.html[/subsystem=messaging-activemq/server/ha-policy=shared-store-secondary] to control the size of the messages transaction block.
+* Add an optional `scale-down-commit-interval` integer to the  https://docs.wildfly.org/35/wildscribe/subsystem/messaging-activemq/server/ha-policy/live-only/index.html[/subsystem=messaging-activemq/server/ha-policy=shared-store-secondary] to control the size of the messages transaction block.
+
+Those attributes shouldn't have any default value.
```

Comment:
Since there is no a default value, what happens when the optional  `scale-down-commit-interval` is not defined?

Reply:
All the messages are transferred to he live node